### ISSUE TITLE
Error trying to open STL file from Solidworks in BEMRosetta EDIT: Fix by running .stl file through online .stl repair

### ISSUE THREAD

**Issue by niilakurkela** [BEMRosetta/BEMRosetta#50]
Hi, 
I'm having the same kind of error as someone else when trying to upload .STL to BEMRosetta for converting it to .dat for NEMOH. I have 15MW VolturnUS-S Semisubmersible floater which I'm trying to import and this error shows up:

''Problem loading 'C:\Users\xxxxxxx\Downloads\jalusta.STL'
file: 'C:\Uxxxx\xxxxxxxx\Downloads\jalusta.STL', line: 2: . 'facet normal' field not found''

This has happened with all STL files from solidworks but then I have converted them in ImageToStl.com to .stl and they have worked but this time it didn't work. 

Please help, I'm kind of in a hurry with this. (Downloaded the file in zip folder since this site doesn't allow .stl files)
[jalusta.zip](https://github.com/user-attachments/files/16506690/jalusta.zip)

**Comment by izabala123** [comment]
Hi Niilakurkela

Thank you for your report. A binary supporting this is just uploaded.
The problem was that, as said by [Wikipedia](https://en.wikipedia.org/wiki/STL_(file_format)), "binary ... STL file has a ... header ... should never begin with ... solid". The uploaded binary stl begins with that, so BEMRosetta supposed that it should have to be STL text. Now BEMRosetta is more permissive.

Apart from that, the STL mesh is not very adequate for BEM calculations. There are huge panels along with small ones. I advise you to get a mesh more regular.

Best regards
Iñaki

**Comment by niilakurkela** [comment]
Thanks for your answer. I'm currently struggling to get results from Nemoh so good to know that the mesh could be the problem. I'm also trying someone else's mesh file now but still struggling. We'll see if I can get it working...
[VOLTURN.zip](https://github.com/user-attachments/files/16509108/VOLTURN.zip)

**Comment by izabala123** [comment]
Hi Niilakurkela

The latter mesh looks much better. Compare for yourself:

![image](https://github.com/user-attachments/assets/3d90f4fb-5618-4033-b78c-d85149214367)
![image](https://github.com/user-attachments/assets/3eeba55e-de9a-44f3-a28e-72646792be7d)

Best regards
Iñaki

**Comment by niilakurkela** [comment]
Yes it looks better, I guess my problem is in using Nemoh now. I don't understand why it doesn't work all of a sudden.. If only someone responded in their gitlab forum so one could ask something there :D 

Thanks for your fast responses Iñaki!

**Comment by izabala123** [comment]
Hi Niilakurkela

I have prepared the case in BEMRosetta an launched in Capytaine. I enclose you the results ([Volturn.zip](https://github.com/user-attachments/files/16510217/Volturn.zip)).
You can play with this case for example adding more headings and frequencies.

Best regards
Iñaki

**Comment by izabala123** [comment]
No answer seems to confirm that the explanation is OK. Issue closed.
